### DISCUSSION TITLE
How to show when I am running a tagged build?

### DISCUSSION THREAD

**Discussion by evandrocoan** [community/community#26447] [Actions]
<p>How can I do something like this:</p>
<pre><code>jobs:
  tests:
    name: ${{ matrix.name }} {{ if startsWith(github.ref, 'refs/tags/') "Tagged" else "" }}
</code></pre>
<p>i.e., when <code>if startsWith(github.ref, 'refs/tags/')</code> is true, my build name should be <code>${{ matrix.name }} Tagged</code>, but when <code>if startsWith(github.ref, 'refs/tags/')</code> is false, my build name should be only <code>${{ matrix.name }}</code>.</p>
<p>I know I could do something like:</p>
<pre><code>jobs:
  tests:
    name: ${{ matrix.name }} startsWith(github.ref, 'refs/tags/')
</code></pre>
<p>Which would make my build names like <code>${{ matrix.name }} true</code> and <code>${{ matrix.name }} false</code>, but this is too verbose and does not auto explain what true and false means.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/evandrocoan">@evandrocoan</a> ,</p>
<p>You can try to add a job to check if the <strong>github.ref</strong> is a <strong>tag</strong> , and set a <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjobs_idoutputs" rel="nofollow noopener"><strong>job-level output</strong></a> to mark the check result, then pass this output to the <strong>name</strong> of the <strong>tests</strong> job.<br>
A simple demo:</p>
<pre><code>jobs:
  check_tag:
    name: Check tag
    outputs:
      tagged: ${{ steps.check_tagged.outputs.is_tagged }}
    runs-on: ubuntu-latest
    steps:
      - name: Check the ref
        id: check_tagged
        run: |
          if [[${{ github.ref }} == refs/tags/*]]; then
            echo "::set-output name=is_tagged::Tagged"
          else
            echo "::set-output name=is_tagged:: "
          fi
  tests:
needs: [check_tag]
    name: ${{ matrix.name }} ${{ needs.check_tag.outputs.tagged }}
    . . .
</code></pre>
<p><strong>Description:</strong></p>
<p><strong>1)</strong> Add the <strong>check_tag</strong> job to check if the <strong>github.ref</strong> is a <strong>tag</strong> : if the ref is a tag, set the value of the <a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-output-parameter" rel="nofollow noopener"><strong>step-level output</strong></a> to be " <strong>Tagged</strong>"; if the ref is not a tag, set the value of the step-level output to be a <strong>whitespace</strong> , of course you also can set it to be others (such as " <strong>Not Tagged</strong>") instead of  <strong>whitespace</strong>. Use the property " <strong>steps.&lt;step id&gt;.outputs.&lt;output name&gt;</strong>" of the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#steps-context" rel="nofollow noopener"><strong>steps context</strong></a> to pass the value of the step-level output to the job-level output.</p>
<p><strong>2)</strong> Set the  <strong>tests</strong> job <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds" rel="nofollow noopener"><strong>needs</strong></a>  <strong>check_tag</strong> job, then you can use the property " <strong>needs.&lt;job id&gt;.outputs.&lt;output name&gt;</strong>" of the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#needs-context" rel="nofollow noopener"><strong>needs context</strong></a> to access the value of the job-level output from the  <strong>check_tag</strong> job.</p>
